### PR TITLE
Implement FemtoFileHandler

### PR DESCRIPTION
**Pull request by leynos**
<h2 dir="auto">Summary</h2>
<ul dir="auto">
<li><strong>New Features</strong>
<ul dir="auto">
<li>Introduced a high-performance file logging handler that writes log records asynchronously to files with thread-safe operation.</li>
<li>Added direct Python interfaces for sending log records to both file and stream handlers.</li>
</ul>
</li>
<li><strong>Bug Fixes</strong>
<ul dir="auto">
<li>Removed fallback to pure Python implementations, ensuring consistent use of the Rust-based extension.</li>
</ul>
</li>
<li><strong>Documentation</strong>
<ul dir="auto">
<li>Updated the roadmap to reflect completed Phase 1 tasks.</li>
</ul>
</li>
<li><strong>Tests</strong>
<ul dir="auto">
<li>Added comprehensive tests for file logging, including concurrent usage scenarios, in both Python and Rust.</li>
</ul>
</li>
</ul>
<h2 dir="auto">Walkthrough</h2>
<p dir="auto">This update introduces a Rust-based <code class="notranslate">FemtoFileHandler</code>
 for asynchronous file logging, integrates it into the Python API, and 
removes the pure Python fallback implementation. The import logic is 
simplified to always use Rust extensions. New tests for the file handler
 are added in both Python and Rust, and the roadmap is updated to mark 
relevant tasks as complete.</p>

------
https://chatgpt.com/codex/tasks/task_e_68607d75a0ac83229bdb16da4937ae6b

## Summary by Sourcery

Remove the pure Python fallback and standardize on the Rust extension; introduce a new asynchronous file-based logging handler and validate it with new tests; update documentation and dependencies accordingly.

New Features:
- Introduce Rust-backed FemtoFileHandler for asynchronous file logging on a background thread.

Enhancements:
- Always import the Rust extension in __init__ and eliminate the pure.py fallback implementation.
- Add a PyO3 binding for `handle` on FemtoStreamHandler.

Build:
- Add `tempfile` as a dev-dependency in Cargo.toml for file handler tests.

Documentation:
- Update roadmap to mark channel integration and the creation of stream and file handlers as completed.

Tests:
- Add Rust unit tests and Python integration tests for the new FemtoFileHandler.

Chores:
- Remove the pure.py module and its imports from the package.

## Summary by Sourcery

Implement a new Rust-based FemtoFileHandler for asynchronous file logging, unify on the Rust extension by removing the pure Python fallback, add Python bindings and extensive tests for file handling, update the roadmap, and clean up obsolete modules

New Features:
- Introduce Rust-backed FemtoFileHandler for asynchronous file logging on a background thread

Bug Fixes:
- Remove fallback to pure Python handler implementations to ensure consistent use of the Rust extension

Enhancements:
- Always import the Rust extension in __init__ and eliminate the pure Python fallback
- Add Python binding for the handle method on FemtoStreamHandler via PyO3

Build:
- Add tempfile as a dev-dependency in Cargo.toml for file handler tests

Documentation:
- Update roadmap to mark MPSC channel integration and handler creation tasks as completed

Tests:
- Add comprehensive Rust and Python tests for FemtoFileHandler covering write behavior, overflow handling, and concurrent usage

Chores:
- Remove pure.py module and its imports from the package

### CODE REVIEW

File: tests/test_file_handler.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    data = path.read_text()
+    for i in range(10):
+        assert f"core [INFO] msg{i}" in data
+
+
+def test_file_handler_open_failure(tmp_path: Path) -> None:
+    bad_dir = tmp_path / "does_not_exist"
+    path = bad_dir / "out.log"
+    with pytest.raises(OSError):
+        FemtoFileHandler(str(path))
```
[the remark this answers is not in the document]
Comment (leynos) [reply tests/test_file_handler.py:57]: That's just silly.

File: tests/test_file_handler.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
+from __future__ import annotations
```
[the remark this answers is not in the document]
Comment (leynos) [reply tests/test_file_handler.py:1]: Docstrings go **above** `from __future__ import annotations`

**Comment by leynos** [comment]
@sourcery-ai review

**Comment by leynos** [comment]
@coderabbitai Please create new GitHub issues for the following:

- Add an explicit close() or flush() method in the Python binding so users can deterministically shut down the background thread and flush pending records without relying solely on GC-triggered Drop.
- Right now every record triggers a file.flush(), which could become a performance bottleneck under heavy load—batch writes or defer flushes until shutdown to reduce I/O overhead (or periodically inject flushes if no logs for n milliseconds).
- Since the bounded channel silently drops records on overflow, we should offer an optional blocking or timeout‐based enqueue mode (or a user hook) so critical logs aren’t lost without notice.

**Comment by leynos** [comment]
@sourcery-ai resolve

**Comment by leynos** [comment]
@coderabbitai review
